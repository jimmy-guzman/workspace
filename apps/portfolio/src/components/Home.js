import React from 'react'
import PropTypes from 'prop-types'

const Home = ({ isEnglish }) => (
  <div className='title'>
    <h1 className='title__name'>
      {isEnglish ? "Hi, I'm Jimmy" : 'Hola, Soy Jimmy'}
    </h1>
    <h2 className='title__caption'>
      {isEnglish ? "I'm a Web Developer" : 'Soy un Desarrollador de Web'}
    </h2>
  </div>
)

Home.propTypes = {
  isEnglish: PropTypes.bool.isRequired
}

export default Home
