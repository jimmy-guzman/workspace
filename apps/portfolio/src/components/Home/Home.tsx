import React from 'react'

export interface HomeProps {
  isEnglish?: boolean
}

export const Home = ({ isEnglish = true }: HomeProps) => (
  <div className='title'>
    <h1 className='title__name'>
      {isEnglish ? "Hi, I'm Jimmy" : 'Hola, Soy Jimmy'}
    </h1>
    <h2 className='title__caption'>
      {isEnglish ? "I'm a Web Developer" : 'Soy un Desarrollador de Web'}
    </h2>
  </div>
)
