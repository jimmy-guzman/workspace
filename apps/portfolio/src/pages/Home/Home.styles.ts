import styled from 'styled-components'
import { flex } from '@portfolio/utils'

export const StyledHomeWrapper = styled.div`
  width: 100%;
  height: 100%;
  ${flex('flex', 'row', 'center', 'center')};
`
