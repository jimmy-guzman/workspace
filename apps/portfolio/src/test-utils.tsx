import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { StyleSheetManager } from 'styled-components'
import { createMemoryHistory } from 'history'

import { ThemeProvider } from './providers'

const Providers: React.ComponentType<{ route?: string }> = ({
  children,
  route,
}: {
  route?: string
  children?: React.ReactNode
}) => {
  const history = createMemoryHistory()

  if (route) {
    history.push(route)
  }

  return (
    <StyleSheetManager disableVendorPrefixes>
      <Router history={history}>
        <ThemeProvider>{children}</ThemeProvider>
      </Router>
    </StyleSheetManager>
  )
}

Providers.displayName = 'ss'

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  route?: string
) => {
  const wrapper: React.ComponentType = ({
    children,
  }: {
    children?: React.ReactNode
  }) => <Providers route={route}>{children}</Providers>

  return render(ui, { wrapper, ...options })
}

export * from '@testing-library/react'

export { customRender as render }
