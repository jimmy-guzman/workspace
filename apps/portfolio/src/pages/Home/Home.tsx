import React from 'react'
import { Title } from '@portfolio/components'
import { homeLang } from '@portfolio/lang'
import { StyledHomeWrapper } from './Home.styles'

export interface HomeProps {
  isEnglish?: boolean
}

export const Home = ({ isEnglish = true }: HomeProps) => (
  <StyledHomeWrapper>
    <Title isEnglish={isEnglish} lang={homeLang.title} />
  </StyledHomeWrapper>
)
