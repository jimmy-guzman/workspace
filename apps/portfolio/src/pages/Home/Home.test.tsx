import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'

import { Home, HomeProps } from './Home'
import { render } from '@portfolio/test-utils'

const setupHome = (props?: HomeProps) => {
  const history = createMemoryHistory()

  return render(
    <Router history={history}>
      <Home {...props} />
    </Router>
  )
}

describe('<Home />', () => {
  it('should render english text', () => {
    const { getByText } = setupHome()

    expect(getByText(/Hi, I'm Jimmy/)).toBeInTheDocument()
    expect(getByText(/I'm a Web Developer/)).toBeInTheDocument()
  })
  it('should render snapshot with not isEnglish', () => {
    const { getByText } = setupHome({ isEnglish: false })

    expect(getByText('Hola, Soy Jimmy')).toBeInTheDocument()
    expect(getByText('Soy un Desarrollador de Web')).toBeInTheDocument()
  })
})
