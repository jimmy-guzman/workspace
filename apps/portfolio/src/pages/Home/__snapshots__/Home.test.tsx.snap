// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render english text 1`] = `
<h1
  class="title__name"
>
  Hi, I'm Jimmy
</h1>
`;

exports[`<Home /> should render english text 2`] = `
<h2
  class="title__caption"
>
  I'm a Web Developer
</h2>
`;

exports[`<Home /> should render snapshot with not isEnglish 1`] = `
<h1
  class="title__name"
>
  Hola, Soy Jimmy
</h1>
`;

exports[`<Home /> should render snapshot with not isEnglish 2`] = `
<h2
  class="title__caption"
>
  Soy un Desarrollador de Web
</h2>
`;
